### PR TITLE
feature/take Photo

### PR DESCRIPTION
**Pull request by Ylanv**
# Description

## Note on PR structure 

This PR description outlines the logic behind the `takePhoto` feature, including elements that will be refined or removed in the next iteration. The final PR description is available [here](https://github.com/PlateSwipe/PlateSwipe/pull/172#issuecomment-2473226339). I've kept this description as-is to document the feature's progression and show how changes were implemented over time. In essence, this approach serves the same purpose as having two separate PRs, but since the initial version was not merged into the main branch due to its unfinished state, I opted to keep the history here. 

Thank you for your understanding!

## **What has been changed?**  
  Added a `takePhoto` method that captures a photo using the provided `ImageCapture` object. This method calls the `onPhotoTaken` callback with the captured image and returns an [`ImageProxy`](https://developer.android.com/reference/androidx/camera/core/ImageProxy) that can be manipulated as needed. 

  Created a `TakePhotoScreen` that displays a camera preview and includes a button for taking a picture. Upon capturing the image, the user is redirected to a new screen where the image is displayed. This screen is currently a placeholder, `DisplayScreen`, but will later be integrated with the final screen in the create recipe flow. Similarly, a temporary `TakePhotoViewModel` has been introduced for demonstration purposes; this will eventually merge into `CreateRecipeViewModel` with an added [`image: Bitmap`](https://developer.android.com/reference/android/graphics/Bitmap) parameter for image storage.

  Additionally, a method was implemented to handle image rotation for images captured in the incorrect orientation. The rotation adjustment is achieved by using [`image.imageInfo.rotationDegrees`](https://developer.android.com/reference/androidx/camera/core/ImageInfo) from the `ImageProxy`.

## **Why was this change made?**  
  These changes were implemented to enable image capture and preview functionality as part of the recipe creation process. This feature is a key element for users to add photos of their recipes directly in-app, enhancing the user experience and supporting the app's visual content.

## **Current Issue ( HALF Solved )** 
There is currently a performance issue where capturing a photo takes abnormally long to process (see attached video for demonstration). This need to be investigated. 

**Solution**: To optimize screen composition timing, consider using [`LaunchedEffect`](https://developer.android.com/reference/kotlin/androidx/compose/runtime/package-summary#LaunchedEffect(kotlin.Any,kotlin.coroutines.SuspendFunction1)) to delay composition until the initial setup is complete. This approach addresses one of the main issues: the screen was re-composing three times, as shown in the log table at the bottom.

By implementing this patch, we can achieve a 1 second improvement in performance. However, an additional 1-second delay remains between the "Button clicked" and "Image Captured" events, which may be challenging to optimize further.
Currently, the user experience may not feel entirely smooth; however, the code is stable and effectively demonstrates the feature's functionality. This implementation can serve as a solid foundation, and I plan to revisit it later for further optimization and refinement.

## Before the Patch 

https://github.com/user-attachments/assets/e709ebec-5655-44f5-902b-989dc59fa7e8

## After the Patch

https://github.com/user-attachments/assets/7d29d95f-4f85-4975-bfad-fe574364d7df

## Checklist

- [x] Tests added.
- [x] Documentation updated.
- [x] All CI checks passed.
- [x] Linked issue/feature (if applicable): #160 

3 times Image Rotated and Screen Display, meaning that `rotateBitmap` was called 3 times.
| Timestamp                | PID    | TID    | Screen                | Package               | Level | Message          |
|--------------------------|--------|--------|-----------------------|-----------------------|-------|-------------------|
| 2024-11-07 14:30:05.506  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Button clicked    |
| 2024-11-07 14:30:06.517  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Image captured    |
| 2024-11-07 14:30:06.649  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | BitMap set        |
| 2024-11-07 14:30:06.649  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | rotation Set      |
| 2024-11-07 14:30:06.691  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Screen Display ON |
| 2024-11-07 14:30:07.302  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Image Rotated     |
| 2024-11-07 14:30:07.433  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Screen Display ON |
| 2024-11-07 14:30:08.086  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Image Rotated     |
| 2024-11-07 14:30:08.201  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Screen Display ON |
| 2024-11-07 14:30:08.832  | 12989  | 12989  | CameraTakePhotoScreen | com.android.sample    | D     | Image Rotated     |

### CODE REVIEW

**Review by AdriB02 — CHANGES_REQUESTED** [review]
Thank you for your work! Your code is easy to read, well-commented, and clean. You've done a great job, especially with the camera functionality! Overall, the code is really solid, with only a few minor details here and there.

I do have one recommendation: I noticed that the test coverage for the `CameraTakePhotoScreen` is a bit low. While you might not be able to test all actions related to taking a photo, I observed that the photo-taking action itself is not currently being tested. You only have to do a `performClick()` action on the button and verify that `navigateTo()` is being called correctly. You may also need to mock some classes for this.

For more details on how to do this, you can refer to the 'Create New Page' wiki here: [Create New Page Wiki](https://github.com/PlateSwipe/PlateSwipe/wiki/Create-new-page#testing).

Keep up the great work!"

**Comment by anaselmadhi** [comment]
Thank you for this detailed pull request. Here’s an in-depth review, covering both positive aspects and suggestions for improvement:

## Positive Aspects
### New Feature for Image Selection:
The addition of GalleryImagePicker.kt is an excellent step to enhance user experience, providing a native way for users to pick images from the gallery. This feature aligns well with modern app design, as users often expect seamless image selection and integration capabilities.
### Testing:
The creation of PhotoPickerTest.kt ensures robust coverage for the new feature, which is critical for identifying any inconsistencies across devices.
The use of TakePhotoViewModelTest to validate state updates (such as bitmap storage and rotation) demonstrates a thorough approach to testing the new components.
### UI and UX Improvements in FridgeScreen:
Repurposing FridgeScreen.kt as a temporary hub for image-related actions provides a flexible environment for demonstration purposes. This is particularly useful for showcasing the new functionalities without needing to build out the entire final structure immediately.
The addition of drawable resources for testing purposes is well-considered and ensures visual accuracy in the tests.
### Integration with Navigation:
Expanding NavigationActions.kt and modifying MainActivity.kt to support the new image selection features makes it easy for users to access the gallery picker.
The approach of adding backward compatibility for older devices in AndroidManifest.xml is thoughtful, ensuring a broader user base can benefit from the feature.
### Modular and Scalable Design:
Splitting functionality across TakePhotoViewModel, GalleryImagePicker, and CameraTakePhotoScreen promotes modular development, making it easier to extend and maintain in future iterations.
The temporary setup in FridgeScreen with clear notes on planned integration helps maintain a focus on modularity while allowing flexible testing.
## Suggestions for Improvement
### Feedback on Image Re-selection Interaction:
The current implementation allows users to click on the image to reopen the picker. While this behavior could be intuitive, it might benefit from a clearer visual indicator, such as an overlay icon or tooltip on the image.
This interaction could also be enhanced by providing an "Edit" or "Change Image" button below the image, explicitly allowing re-selection. This approach could offer clearer guidance, especially for first-time users.
### Potential Refactor of FridgeScreen:
Given that FridgeScreen is a temporary hub, consider adding comments to indicate that this layout and functionality are placeholders. This helps maintain clarity in code review, especially when multiple contributors are involved.
Once integrated, it would be useful to either refactor FridgeScreen to reduce coupling or replace it with a more modular solution.
### Documentation Enhancement:
Although your notes in the PR description are thorough, a brief inline comment within TakePhotoViewModel and CameraTakePhotoScreen about the temporary nature of these implementations would be helpful for future developers.
Consider adding high-level comments in PhotoPickerTest.kt outlining the specific test scenarios covered. This documentation could serve as a quick reference for understanding testing coverage.
Improving Compatibility Setup in AndroidManifest.xml:
While the manifest configuration enables automatic installation of the backported PhotoPicker module, including a note within the manifest or the PR description on how it may affect APK size or initialization could be beneficial.
It may also be worth explicitly mentioning in the code or PR whether there are any fallbacks for users on older devices who may not receive the backport module.
### Edge Case Handling in rotateBitmap Tests:
The tests for rotateBitmap could include edge cases for unusual rotation degrees, such as non-90-degree multiples. This could further validate the method’s robustness and preempt potential future issues, especially if user-captured images have unexpected metadata.
### Potential UI Enhancement:
The visual consistency between the TakePhotoButton and the image display screen could benefit from further polish. Consider applying a similar style (e.g., colors, shape) for interactive buttons or icons on FridgeScreen and DisplayImageScreen to maintain uniformity.
## Final Feedback
Overall, this PR is very well-organized and thought out. The addition of image-picking functionality is valuable, and your testing approach is comprehensive. Moving forward, the suggestions outlined above should help enhance usability and clarity, ensuring the feature is both intuitive and maintainable.
Your attention to temporary setups, backward compatibility, and modular design is excellent and showcases best practices. Great work on this PR!

**Comment by Ylanv** [comment]
Thank you for the detailed feedback! I noticed you referenced `GalleryImagePicker.kt`, which was actually implemented in PR #174 , not in this one. 

As for your suggestion on adding comments for temporary code, I agree it’s a helpful idea. In this case, however, the code will soon be updated to a final version in the next commit, but I’ll keep this approach in mind for future cases.

Regarding your point on handling edge cases for `rotateBitmap`, are you referring to testing this functionality? If so, there’s already a test for this: 

```kotlin
@Test
fun testRotateBitmap_multipleOf90() {
    val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)
    assertThrows(AssertionError::class.java) {
        rotateBitmap(bitmap, 45) // Should fail
    }
    assertThrows(AssertionError::class.java) {
        rotateBitmap(bitmap, -45) // Should fail
    }
}
```

Or were you asking why `rotateBitmap` only accepts multiples of 90 degrees? Let me know if I can clarify further.

**Comment by Ylanv** [comment]
# Description
## **What has been changed?**  
   This PR integrates the take-photo functionality with the broader `createRecipe` process by adding the `RecipeAddImageScreen.kt` screen. This new screen allows users to either take a picture or import a photo from their gallery (note: gallery import is pending in PR #174 ). Additionally, the `TakePhotoViewModel` has been removed, and its functionality has been merged into `CreateRecipeViewModel` to streamline the code. Specifically, the `setBitmap` and `setRotation` methods have been combined for clarity.  
   
Now, users must upload an image to proceed with the recipe creation process, ensuring that each recipe has an associated image before publishing (see attached video for demonstration). Additionally, the recipe `UID` is reused for the image `UID`, streamlining the process by avoiding the need to generate a separate `UID` for the image

##  **Why was this change made?**  
This enhancement provides a more cohesive and user-friendly experience when creating recipes, allowing users to add photos directly. It also standardizes the creation process by storing images in Firebase Storage under `test/RECIPE_UID`, which enables easier testing without impacting the production database.

## **What could be improve** 
For this section, I’m sharing my thoughts on areas for improvement that could be beneficial for future iterations. I’d really appreciate it if you could also contribute your thoughts and suggestions:

1. **Refining the UI**: Currently, the UI is functional but feels somewhat lacking in character. Adding some visual enhancements could give it a more polished and engaging feel.
  
2. **Improving Photo Capture Time**: As noted earlier in the PR, the time it takes to capture a photo is currently quite long. Further investigation is needed to determine if this can be optimized, and any suggestions or insights would be welcome.

Your feedback on these points would be valuable as we work towards refining this feature.

## Checklist
- [x] Tests added.
- [x] Documentation updated.
- [x] All CI checks passed.
- [x] Linked issue/feature (if applicable): #160 

As noted above, this is the final PR summary to maintain a record of progress. Please feel free to test recipe creation within the Firestore `test` collection to avoid any impact on the production database. Once testing is complete, please notify me, and I will reconfigure it to the production database and clean up the `test` folder.

## Demonstration Video

https://github.com/user-attachments/assets/6d11ee35-156e-4ea4-b47b-160526a61462

## Image storage after recipe is published
image

## Recipe Storage after recipe is published
image

File: app/src/androidTest/java/com/android/sample/createRecipe/RecipeNameScreenTest.kt [hunk 1]
```diff
@@ -31,7 +34,8 @@ class RecipeNameScreenTest {
     // Mock Firestore and Repository for use in ViewModel
     val firestore = mockk<FirebaseFirestore>(relaxed = true)
     val repository = FirestoreRecipesRepository(firestore)
-    createRecipeViewModel = spyk(CreateRecipeViewModel(repository)) // Spy to verify calls
+    repoImg = mockk(relaxed = true)
+    createRecipeViewModel = spyk(CreateRecipeViewModel(repository, repoImg)) // Spy to verify calls
 
     // Set the content for testing
```
Comment (AdriB02) [inline app/src/androidTest/java/com/android/sample/createRecipe/RecipeNameScreenTest.kt:41]: The `tearDown()` method currently does nothing, as there are no resources being released or clean-up actions required. Since it's not serving any purpose, I suggest deleting it to keep the code cleaner and more concise.

File: app/src/main/java/com/android/sample/resources/C.kt [hunk 2]
```diff
@@ -147,7 +150,7 @@ object C {
     const val MEAL_DB_CATEGORY_ARRAY = "categories"
 
     // FirestoreRecipeRepository
-    const val FIRESTORE_COLLECTION_NAME = "recipes"
+    const val FIRESTORE_COLLECTION_NAME = "test"
```
Comment (AdriB02) [inline app/src/main/java/com/android/sample/resources/C.kt:153]: It looks like you may have missed removing the test collection name. It would be better to use the recipes collection from Firestore instead.
```suggestion
    const val FIRESTORE_COLLECTION_NAME = "recipes"
```

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt
Comment (AdriB02) [inline app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt:1]: This UI is absolutely fantastic! It’s beautifully simple and clean, with just the right amount of information, nothing overwhelming. I love how intuitive and sleek it looks! There's just one small bug on smaller phones: the cook image gets hidden by the button. Maybe you could slightly reduce the button size on smaller screens and give the cook image a bit more prominence by increasing its weight. This would help keep the design balanced and enhance the user experience even more!
WhatsApp Image 2024-11-13 à 21 27 59_2dacaa14

**Review by AdriB02 — CHANGES_REQUESTED** [review]
You did a really great job, beautiful UIs and a lot of solid backend work! Thanks a lot for your efforts. The code is easy to read, and the comments make everything clear, so there are no issues understanding the purpose of each part. I appreciate the clean structure and thoughtful design.

There are a few small bugs and improvements that I recommended, but nothing too major. These tweaks will help polish the code and improve the overall experience, but overall, everything is well-structured and functional.

Thanks again for your hard work, it's truly appreciated! Keep up the excellent work!

File: app/src/main/java/com/android/sample/resources/C.kt (hunk 2 above)
Comment (Ylanv) [reply app/src/main/java/com/android/sample/resources/C.kt:153]: Thanks, for the catch up i will switch it to "recipes with images" which is now the official database.

File: app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt [hunk 3]
```diff
@@ -247,18 +278,34 @@ class CreateRecipeViewModel(private val repository: FirestoreRecipesRepository)
     val newUid = repository.getNewUid()
     require(newUid.isNotBlank()) { "Recipe ID must not be blank." }
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt:279]: After some reflection, it appears this check might be unnecessary. Since `repository.getNewUid()` calls `db.collection(FIRESTORE_COLLECTION_NAME).document().id`, which, according to the documentation, reliably returns a non-null object, we can safely remove this check.

File: app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt [hunk 4]
```diff
@@ -247,18 +278,34 @@ class CreateRecipeViewModel(private val repository: FirestoreRecipesRepository)
...
+                  onFailure = { exception ->
+                    _publishStatus.value = RECIPE_PUBLISH_ERROR_MESSAGE.format(exception.message)
+                  })
+            },
+            onFailure = {
+              // Throw an error if the image upload fails
+              throw IllegalArgumentException("Image upload failed.")
+            })
+      } else {
+        throw IllegalArgumentException("Image must not be blank.")
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt:307]: Yes, I made two changes. First, if the photo is null, this will now throw a `NullPointerException` rather than the previous `IllegalArgumentException`. Second, when an `onFailure` callback is invoked, the error handling is now consistent across both callbacks by setting `_publishStatus.value = RECIPE_PUBLISH_ERROR_MESSAGE.format(exception.message)`. This approach ensures a uniform error-handling process.

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+@Composable
+fun AddImageContent(
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel
+) {
+  var isPictureTaken by remember { mutableStateOf(false) }
+  val context = LocalContext.current
+  // Collect the photo bitmap from the ViewModel's StateFlow
+  val bitmap: Bitmap? by createRecipeViewModel.photo.collectAsState()
+  bitmap?.let { isPictureTaken = true }
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt:126]: Great, Thanks you!

File: app/src/main/java/com/android/sample/ui/camera/CameraTakePhotoScreen.kt [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                  // Set the captured photo and its rotation in the ViewModel
+                  takePhotoViewModel.setBitmap(image.toBitmap())
+                  takePhotoViewModel.setRotation(image.imageInfo.rotationDegrees)
+                  // Navigate to the display image screen
+                  navigationActions.navigateTo(Screen.DISPLAY_IMAGE)
+                })
+          }
+        }
+      })
+}
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/camera/CameraTakePhotoScreen.kt:71]: Thank you for your suggestions. For the first point, I’m having some difficulty seeing the added value of including that design element, as the `CameraPreview` currently follows the standard design used in the native Android camera app. However, I’ll conduct some user testing to see if this feedback comes up frequently. Regarding the second point, adding zoom functionality would indeed be a great enhancement. At this stage, though, implementing it is too time-consuming, so I’ll plan to incorporate it in the next iteration.

File: app/src/main/java/com/android/sample/model/recipe/RecipeBuilder.kt [hunk 7]
```diff
@@ -135,7 +136,7 @@ class RecipeBuilder {
       "Recipe instructions are required and cannot be blank."
     }
     require(ingredientsAndMeasurements.isNotEmpty()) { "At least one ingredient is required." }
-
+    require(strMealThumbUrl.isNotBlank()) { "Recipe thumbnail is required and cannot be blank." }
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/model/recipe/RecipeBuilder.kt:139]: I didn't implement this part but i will do it.

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+                        Modifier.fillMaxWidth()
+                            .height(
+                                (C.Dimension.RecipeAddImageScreen.TEXT_HEIGHT *
+                                        LocalConfiguration.current.screenHeightDp)
+                                    .dp)
+                            .testTag("title col"),
+                    horizontalAlignment = Alignment.CenterHorizontally) {
+                      Text(
+                          text = stringResource(R.string.add_image),
+                          style = MaterialTheme.typography.h5,
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt:156]: I decided to remove the small text below, as 'Add Image' is already quite straightforward. I’ll update the Figma file to reflect this change.

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt:1]: To align with the work of the other developper, I used shouldDisplayChefImage() to know if the phone is big enough to display it.

File: app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (AdriB02) [reply app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt]: Thanks, @Ylanv! Moving to `NullPointerException` for a null photo makes sense, and I appreciate the uniform error-handling setup. Moving the error message to the constants file now, and will check the updated callback logic. Thanks for making those adjustments!

File: app/src/main/java/com/android/sample/resources/C.kt (hunk 2 above)
Comment (AdriB02) [reply app/src/main/java/com/android/sample/resources/C.kt]: Thanks for letting us know, @Ylanv! Switching to 'recipes with images' makes sense, and we appreciate the heads-up about the temporary black images on SwipePage, this helps us stay informed. Looking forward to the next PR!

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt
Comment (AdriB02) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt:1]: Great, I see you’ve fixed it, thanks a lot!

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+                                  shape = RoundedCornerShape(size = C.Dimension.PADDING_4.dp)),
+                      shape = RoundedCornerShape(C.Dimension.PADDING_4.dp),
+                      enabled = isPictureTaken) {
+                        Text(
+                            text = stringResource(R.string.next),
+                            style = MaterialTheme.typography.button)
+                      }
+                }
+              }
+        }
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt:299]: Done

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+                                contentDescription = stringResource(R.string.camera),
+                                modifier =
+                                    Modifier.size(
+                                        (C.Dimension.RecipeAddImageScreen.ICON_SIZE *
+                                                LocalConfiguration.current.screenWidthDp)
+                                            .dp),
+                                tint = Color.Black)
+                            Text(
+                                text = stringResource(R.string.camera),
+                                style = MaterialTheme.typography.body1)
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt]: Done

File: app/src/main/java/com/android/sample/ui/camera/CameraTakePhotoScreen.kt [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+/**
+ * Composable function to display the camera screen and handle photo capture.
+ *
+ * @param navigationActions Actions to navigate between screens.
+ * @param createRecipeViewModel ViewModel to manage the state of the photo and its rotation.
+ */
+@androidx.annotation.OptIn(ExperimentalGetImage::class)
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun CameraTakePhotoScreen(
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/camera/CameraTakePhotoScreen.kt:40]: I answered in the other [comment](https://github.com/PlateSwipe/PlateSwipe/pull/172/files#r1833200119) :)

File: app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+                          Modifier.width(RECIPE_NAME_BUTTON_WIDTH)
+                              .height(RECIPE_NAME_BUTTON_HEIGHT)
+                              .background(
+                                  color = lightCream,
+                                  shape = RoundedCornerShape(size = C.Dimension.PADDING_4.dp)),
+                      shape = RoundedCornerShape(C.Dimension.PADDING_4.dp),
+                      enabled = isPictureTaken) {
+                        Text(
+                            text = stringResource(R.string.next),
+                            style = MaterialTheme.typography.button)
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/main/java/com/android/sample/ui/createRecipe/RecipeAddImageScreen.kt]: Done!

File: app/src/androidTest/java/com/android/sample/createRecipe/AddInstructionStepContentTest.kt [hunk 13]
```diff
@@ -26,14 +27,16 @@ class AddInstructionStepScreenTest {
...
 
   @get:Rule val composeTestRule = createComposeRule()
 
   @Before
   fun setUp() {
+    repoImg = mockk(relaxed = true)
     navigationActions = mockk(relaxed = true)
-    createRecipeViewModel = spyk(CreateRecipeViewModel(repository))
+    createRecipeViewModel = spyk(CreateRecipeViewModel(repository, repoImg))
     Intents.init()
```
[the remark this answers is not in the document]
Comment (Ylanv) [reply app/src/androidTest/java/com/android/sample/createRecipe/AddInstructionStepContentTest.kt]: Removed!

**Review by AdriB02 — APPROVED** [review]
Thank you for implementing the modifications! The updates look spot-on, and everything is working smoothly. The code is clear, and the adjustments improve readability and functionality. Everything looks good to merge. Great job!
